pkg/smokescreen/smokescreen_test.go

```diff
@@ -1069,7 +1186,7 @@ func proxyServer(conf *Config) *httptest.Server {
 	return httptest.NewServer(proxy)
 }
 
-func proxyClient(proxy string) (*http.Client, error) {
+func proxyClient(proxy string, h http.Header) (*http.Client, error) {
```

Comment:
(Optional suggestion) `h` or `header` might be too general. Maybe we can call it `proxyHeader` or `proxyConnectHeader`?
```suggestion
func proxyClient(proxy string, proxyConnectHeader http.Header) (*http.Client, error) {
```

On the other hand, considering that we use this header only in one place, can/should we leave this func as-is and update the client instead of passing `nil` everywhere?
```go
client, err := proxyClient(proxy.URL)
// check the err
client.Transport.ProxyConnectHeader = h
```

Just FYI, for our internal use case we're going to pass 2 extra headers instead of one extra header. So if we were to rename the variable, maybe it would be better to underscore that it could be plural.

Reply:
The types are a bit problematic here. `client.Transport` is typed as `RoundTripper`, not `http.Transport`, so what you suggest leads to a compile error (unless I'm misunderstanding something). I'll rename it to `proxyConnectHeaders` for clarity.